rest-server/engines/src/main/scala/com/actionml/engines/ur/URDataset.scala

```diff
@@ -176,39 +176,38 @@ class URDataset(engineId: String, val store: Store) extends Dataset[UREvent](eng
     items.foreach(insertProperty)
   }
 
-  private def insertProperty(event: UREvent): Unit =
-    (for {
-      updateItem <- itemsDao.findOneByIdAsync(event.entityId).map(_.getOrElse {
-        logger.debug(s"No item found with id ${event.entityId}")
-        URItemProperties(event.entityId, Map.empty)
-      })
-      _ <- itemsDao.saveOneByIdAsync(
-        event.entityId,
-        URItemProperties(
-          _id = updateItem._id,
-          dateProps = updateItem.dateProps ++ event.dateProps,
-          categoricalProps = updateItem.categoricalProps ++ event.categoricalProps,
-          floatProps = updateItem.floatProps ++ event.floatProps,
-          booleanProps = updateItem.booleanProps ++ event.booleanProps
-        )
+  private def insertProperty(event: UREvent): Unit = {
+    val updateItem = itemsDao.findOneById(event.entityId).getOrElse {
+      logger.debug(s"No item found with id ${event.entityId}")
+      URItemProperties(event.entityId, Map.empty)
+    }
+    itemsDao.saveOneById(
+      event.entityId,
+      URItemProperties(
+        _id = updateItem._id,
+        dateProps = updateItem.dateProps ++ event.dateProps,
+        categoricalProps = updateItem.categoricalProps ++ event.categoricalProps,
+        floatProps = updateItem.floatProps ++ event.floatProps,
+        booleanProps = updateItem.booleanProps ++ event.booleanProps
       )
-      newProps = (
-        event.categoricalProps.mapValues(_.toList) ++
+    )
+    val newProps = (
+      event.categoricalProps.mapValues(_.toList) ++
         event.booleanProps.mapValues(_.toString :: Nil) ++
         event.dateProps.mapValues { d => writeFormat.format(d.toInstant) :: Nil } ++
         event.floatProps.mapValues(_.toString :: Nil)
       ).filterNot {
-        case (name, _) => indicatorNames.contains(name)
-      }
-      esDoc = {
-        val doc = es.findDocById(event.entityId)
-        (doc._1, doc._2 ++ newProps)
-      }
-      result <- es.saveOneByIdAsync(event.entityId, esDoc)
-    } yield logger.trace(s"Document $esDoc ${if (result) " successfully saved" else " failed to save"} to Elastic Search")
-    ).recover {
-      case NonFatal(e) => logger.error(s"Engine-id: ${engineId}. Can't insert item $event", e)
+      case (name, _) => indicatorNames.contains(name)
     }
+    (for {
+      doc <- es.findDocByIdAsync(event.entityId)
```

Comment:
this may not find a doc, it that ok here?

$set should create a mongo or es doc if one does not exist, or it should merge $set values with the existing doc. There is no failure mode for this unless the write operation fails.

Reply:
this `findDocByIdAsync` method return empty result if it can't find by entityId